src/api_server/src/session.rs

```diff
@@ -104,13 +113,63 @@ impl<'a> Session<'a> {
         self.is_authenticated() && !self.is_expired()
     }
 
+    pub fn attestation_results(&self) -> Option<AttestationResults> {
+        self.attestation_results.clone()
+    }
+
     pub fn set_attestation_results(&mut self, attestation_results: AttestationResults) {
         self.attestation_results = Some(attestation_results)
     }
 
     pub fn set_tee_public_key(&mut self, key: TeePubKey) {
         self.tee_pub_key = Some(key)
     }
+
+    pub fn to_jwe(&self, payload_data: Vec<u8>) -> Result<Response> {
+        let tee_pub_key = self
+            .tee_public_key()
+            .ok_or_else(|| anyhow!("No TEE public Key"))?;
+        if tee_pub_key.alg != *RSA_ALGORITHM {
+            return Err(anyhow!("Unsupported TEE Pub Key type or algorithm"));
+        }
+
+        let mut rng = rand::thread_rng();
+
+        let aes_sym_key = Aes256Gcm::generate_key(&mut OsRng);
+        let cipher = Aes256Gcm::new(&aes_sym_key);
+        let iv = rng.gen::<[u8; 12]>();
+        let nonce = Nonce::from_slice(&iv);
+        let encrypted_payload_data = cipher
+            .encrypt(nonce, payload_data.as_slice())
+            .map_err(|e| anyhow!("AES encrypt Resource payload failed: {:?}", e))?;
+
+        let k_mod = base64::decode(&tee_pub_key.k_mod)?;
+        let n = BigUint::from_bytes_be(&k_mod);
+        let k_exp = base64::decode(&tee_pub_key.k_exp)?;
+        let e = BigUint::from_bytes_be(&k_exp);
+
+        let rsa_pub_key = RsaPublicKey::new(n, e)
+            .map_err(|e| anyhow!("Building RSA key from modulus and exponent failed: {:?}", e))?;
+        let padding = PaddingScheme::new_pkcs1v15_encrypt();
+        let sym_key: &[u8] = aes_sym_key.as_slice();
+        let wrapped_sym_key = rsa_pub_key
+            .encrypt(&mut rng, padding, sym_key)
+            .map_err(|e| anyhow!("RSA encrypt sym key failed: {:?}", e))?;
+
+        let protected_header = json!(
+        {
+           "alg": RSA_ALGORITHM.to_string(),
+           "enc": AES_GCM_256_ALGORITHM.to_string(),
+        });
+
+        Ok(Response {
+            protected: serde_json::to_string(&protected_header)?,
```

Comment:
I don't have much experience in this area, will the header here be unsafe? Specifically, how should it be encrypted?

Additionally, this is not the scope of this PR, we can create a new PR to fix it after this PR is merged.

@sameo What do you think?

Reply:
The JWE spec defines that the "protected" field should be integrity protected. This is "additional authenticated data" in AES-GCM, the resulting tag goes in the "tag" field. It does not need to be encrypted.

Otherwise we the field should be called "unprotected".

I'm OK if we address this later. This will be a breaking change.